src/Framework/Framework/Resources/Scripts/validation/validation.ts

```diff
@@ -44,26 +44,48 @@ const createValidationHandler = (path: string) => ({
             options.validationTargetPath = path;
             // resolve target
             const context = ko.contextFor(options.sender);
-            const validationTarget = evaluator.evaluateOnViewModel(context, path);
+            const validationTarget = evaluator.traverseContext(context, path);
 
-            watchAndTriggerValidationErrorChanged(options, () => {
-                detachAllErrors();
-                validateViewModel(validationTarget);
-            });
+            runClientSideValidation(validationTarget, options);
 
             if (allErrors.length > 0) {
                 logError("validation", "Validation failed: postback aborted; errors: ", allErrors);
                 return Promise.reject(new DotvvmPostbackError({ type: "handler", handlerName: "validation", message: "Validation failed" }))
             }
         }
-        return callback()
+        return callback();
+    }
+})
+
+const runClientSideValidation = (validationTarget:any,options:PostbackOptions) => {
+
+    watchAndTriggerValidationErrorChanged(options,
+        () => {
+            detachAllErrors();
+            validateViewModel(validationTarget);
+        });
+}
+
+const createRootValidationHandler = () => ({
```

Comment:
Is this used anywhere?

Reply:
It is used to create a handler for `validate-root` that was just resurrected 😄.